app/controllers/api/v0/reactions_controller.rb

```diff
@@ -1,25 +1,24 @@
 module Api
   module V0
-    class ReactionsController < ApplicationController
+    class ReactionsController < ApiController
       skip_before_action :verify_authenticity_token
 
       def create
-        @user = valid_user
-
-        unless @user
-          render json: { message: "invalid_user" }, status: :unprocessable_entity
+        reaction_user = load_reaction_user
+        unless reaction_user
+          render json: { error: "invalid user" }, status: :unprocessable_entity
           return
         end
 
-        Rails.cache.delete "count_for_reactable-#{params[:reactable_type]}-#{params[:reactable_id]}"
-
-        @reaction = Reaction.create(
-          user_id: @user.id,
-          reactable_id: params[:reactable_id],
-          reactable_type: params[:reactable_type],
-          category: params[:category] || "like",
+        @reaction = Reaction.create!(
+          user: reaction_user,
+          reactable_id: reaction_params[:reactable_id],
+          reactable_type: reaction_params[:reactable_type],
+          category: reaction_params[:category] || "like",
         )
 
+        delete_reactable_cache(reaction_params[:reactable_id], reaction_params[:reactable_type])
```

Comment:
I figured it makes more sense NOT to delete the cache if the reaction is not valid :)